client/src/components/editor/hooks/UseContextMenu.tsx

```diff
@@ -0,0 +1,37 @@
+import {useState} from "react";
+import {ViewportPoint, getViewportBounds} from "../Util";
+
+
+interface UseContextMenuResult {
+    isOpen: boolean;
+    anchor: ViewportPoint;
+    anchorAt: (baseAnchor: ViewportPoint, menu: HTMLElement) => void;
+    close: () => void;
+}
+
+function useContextMenu() : UseContextMenuResult {
+    const [isOpen, setIsOpen] = useState(false);
+    const [anchor, setAnchor] = useState<ViewportPoint>({x: 0, y: 0});
+
+    const anchorAt = (baseAnchor: ViewportPoint,  menu: HTMLElement) => {
+        const vh = document.documentElement.clientHeight;
```

Comment:
Went with the solution of calculating the height of the context menu and adjusting its anchor so that remains fully visible with the viewport instead of hardcoding a height.